apollo-common/src/main/java/com/ctrip/framework/apollo/common/exception/AbstractApolloHttpException.java

```diff
@@ -24,8 +25,15 @@
   
   protected HttpStatus httpStatus;
 
-  public AbstractApolloHttpException(String msg){
-    super(msg);
+  /**
+   * When args not empty, use {@link com.google.common.base.Strings#lenientFormat(String, Object...)}
+   * to replace %s in msgtpl with args to set the error message. Otherwise, use msgtpl 
+   * to set the error message. e.g.: 
+   * <pre>{@code new NotFoundException("... %s ... %s ... %s", "str", 0, 0.1)}</pre>
+   * If the number of '%s' in `msgtpl` does not match args length, the '%s' string will be printed.
+   */
+  public AbstractApolloHttpException(String msgtpl, Object... args){
+    super(args == null || args.length == 0 ? msgtpl : Strings.lenientFormat(msgtpl, args));
```

Comment:
Why do we use `Strings.lenientFormat(msgtpl, args)` instead of `String.format(msgtpl, args)` here?

Reply:
`Strings.lenientFormat(..)` performs better than `String.format(..)`. `Strings.lenientFormat(..)` uses `StringBuilder` to handle strings, `String.format(..)` uses regex to handle strings.

```
String appId = "app-1001";
String clusterName = "test";
String namespaceName = "application";
String key = "test.key";

long start = System.currentTimeMillis();
for (int i = 0; i < 1000; i++) {
  String s = Strings.lenientFormat("item not found for %s %s %s %s", appId, clusterName, namespaceName, key);
}
System.out.println("Strings.lenientFormat called 1000 times, used: " + (System.currentTimeMillis() - start) + "ms");

start = System.currentTimeMillis();
for (int i = 0; i < 1000; i++) {
  String s = String.format("item not found for %s %s %s %s", appId, clusterName, namespaceName, key);
}
System.out.println("String.format called 1000 times, used: " + (System.currentTimeMillis() - start) + "ms");
```
Running this case will print out:
Strings.lenientFormat called 1000 times, used: 4ms
String.format called 1000 times, used: 29ms